From 9f7740c4c2fda64029c23674e9858ce8bd4367df Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 13 Jul 2017 17:14:05 -0700
Subject: [PATCH] util.h: include <sys/reg.h> when libc != glibc

For musl libc it is required to include <sys/reg.h> to
have __WORDSIZE defined to e.g. 32 for arm*-musl.

Taken from void-linux
https://github.com/voidlinux/void-packages/blob/master/srcpkgs/dracut/patches/musl-__wordsize.patch

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/install/util.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/install/util.h b/src/install/util.h
index a6f9a184..82732913 100644
--- a/src/install/util.h
+++ b/src/install/util.h
@@ -36,6 +36,9 @@
 #include <sys/stat.h>
 #include <dirent.h>
 #include <sys/resource.h>
+#if !defined(__GLIBC__)
+#include <sys/reg.h>
+#endif
 
 #include "macro.h"
 
-- 
2.17.1

